core/src/main/java/com/linecorp/armeria/client/ResponseAs.java

```diff
@@ -140,7 +142,44 @@ static <T> FutureResponseAs<ResponseEntity<T>> json(TypeReference<? extends T> t
                                                         ObjectMapper mapper) {
         requireNonNull(typeRef, "typeRef");
         requireNonNull(mapper, "mapper");
-        return aggregateAndConvert(AggregatedResponseAs.json(typeRef, mapper));
+        return aggregateAndConvert(AggregatedResponseAs.json(typeRef, mapper, SUCCESS_PREDICATE));
+    }
+
+    @UnstableApi
+    static <V> BlockingConditionalResponseAs<V> andThenJson(
```

Comment:
Users can use `json` as just same as before.
On the other hand, they also can use `andThenJson` in case they want to apply a mapping by several conditions like this.
```java
ResponseAs.<MyResponse>andThenJson(MyMessage.class, res -> res.status().isServerError())
                                       .andThenJson(MyMessage.class, res -> res.status().isClientError())
                                       .orElseJson(MyMessage.class)
```

In current implementation, I couldn't intend to chain `andThenJson` after `json`.

Reply:
Oh, I just meant that I think `ResponseAs#andThenJson` can be renamed to `ResponseAs#json` (this would mean `ResponseAs#json` would be overloaded depending on the predicate parameter).

The reasoning was that uses can type `ResponseAs.json(` and auto-complete suggests the API, which results in more exposure. Also, users don't have to remember `ResponseAs.andThenJson()` and can just type `.json()`.
This is a minor suggestion though 😄